prototypes/ScheduledMerges.hs

```diff
@@ -1158,9 +1175,44 @@ depositNominalCredit (NominalDebt nominalDebt)
 -- Updates
 --
 
+-- | At some point, we want to merge the union level with the regular levels.
+-- We achieve this by moving it into a new last regular level once it is both
+-- completed (merged down to a single run) and fits into such a new level.
+--
+-- Our representation doesn't allow for empty levels, so we can only put the
+-- run directly after the pre-existing regular levels. If it is too large for
+-- that, we don't want to move it yet to avoid violating run size invariants
+-- and doing inefficient merges of runs with very different sizes.
```

Comment:
Maybe we should make a TODO to allow empty levels? Or maybe having the `Single` vs. `MigratedUnion` distinction would help with thiss?

Reply:
I just realised I originally wanted to do something even simpler: When the union is completed, always move it to a new level. If it is much larger than that level should be, the existing code will already handle, not creating a merge with it, but just pushing the oversized run down the levels over time, until it fits in and becomes part of a new last level merge. I think combined with the `MigratedUnion` constructor (to avoid watering down the invariant too much), that could be a decent solution. Kind of like allowing empty levels just before a `MigratedUnion`, but not explicitly representing them.

Reply:
If so, maybe we should have a dedicated test that triggers this particular behaviour so that we can check that it works correctly